### ISSUE TITLE
Rework of logging functions

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#83]
The following is taken out of already closed
https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/67
(because issue 67 contained too much aspects)

We have two methods with different functionality:

`logger.error(<error message>)` and `BuiltIn().unknown(<error message>)`

And the first one prints the error message starting with red [ERROR] string. The second one makes the test result UNKNOWN and also provides the possibility to define an error message.

We are in situations where we need to call both functions, but we do not want to have the error message printed twice. Therefore only one of the function calls should contain this error message - maybe the `logger.error()`. Then `BuiltIn().unknown()` is called without error message.

Unfortunately this does not work properly.

Because `logger.error(<error message>) `does not write the error message into the text log file (-b command line option) - but `BuiltIn().unknown(<error message>)` does. We do not have the error message twice any more in console, but we also do not have the error message in text log file any more. In my opinion this is not acceptable.

And also `BuiltIn().unknown()` has it's own issue: Already several times it was annoying for me to see this message in the text log file:
   `UNKNOWN - Unknown exception occurs. Details: None`
An exception without any details about it? Sounds worryingly for me.

In RobotFramework\python39\Lib\site-packages\robotframework-4.1.3-py3.9.egg\robot\errors.py I found the reason:
```
   def _get_message(self):
       msg = "Unknown exception occurs. Details: %s" % self.msg
       return msg
```
With `self.msg` initialized with `None`.

I do not like this implementation! The default value makes the error message an option! And in case of a user decides not to provide an error message when he uses `BuiltIn().unknown()` - for what reason ever - then the function has to be quiet! Then the function has to set the test result to UNKNOWN, nothing else.

In my opinion the `self.msg` has to be checked against `None` and "". And only in case of a user provides something, the function is allowed to add "`Unknown exception occurs. Details: `" to the message. Otherwise `self.msg` should be returned unchanged.

But this also does not solve the problem mentioned above. We need a strict separation of concern:
* A function only for printing an error message, but to every possible destination (console and all log files) and without doing anything else
* A function influencing the test result (with maybe printing also an error message, but only as option)

**Comment by namsonx** [comment]
Hello Holger,

I can summary your comment into two main issues you currently face:

1. The same error message occurs twice in console log.
2. Unknow status with reason "Details: None"

For the second issue "Unknow status with reason 'Details: None", we can discuss with Thomas then come with final decision. This is related to Robot Framework core and @milanac030988 will create a change.

For the first issue: From my point of view, it makes sense to have error message at beginning of console log then this error message again is displayed at the end of console log to inform users that all testsuites are in "Unknow" status due to this error message. In reality, a console log is quite long it depends to the number of test cases, for example please take a look into the console log below.
.. code::
      ==============================================================================
      Testsuites
      ==============================================================================
      [ ERROR ] Testsuite management - Loading configuration level 2 failed!
                       The variant 'default' is not defined in 'C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\robotframework-tutorial\900_Building_testsuites\exercise-08\config\exercise-08_variants.json'
      [ ERROR ] Did not find configuration file: ''!
      Testsuites.component-A
      ==============================================================================
      Testsuites.component-A.feature-A-1
      ==============================================================================
      Testsuites.component-A.feature-A-1.exercise-08-feature-A-1-test
      ==============================================================================
      Test Case exercise-08-feature-A-1-test :: exercise-08-feature-A-1-... | UNKNOWN |
      Variable '${teststring_common}' not found.
      ------------------------------------------------------------------------------   
      Testsuites.component-A.feature-A-1.exercise-08-feature-A-1-test       | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-A.feature-A-1                                    | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-A.feature-A-2
      ==============================================================================
      Testsuites.component-A.feature-A-2.exercise-08-feature-A-2-test
      ==============================================================================
      Test Case exercise-08-feature-A-2-test :: exercise-08-feature-A-2-... | UNKNOWN |
      Variable '${teststring_common}' not found.
      ------------------------------------------------------------------------------
      Testsuites.component-A.feature-A-2.exercise-08-feature-A-2-test       | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-A.feature-A-2                                    | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-A                                                | UNKNOWN |
      2 tests, 0 passed, 0 failed, 2 unknown
      ==============================================================================
      Testsuites.component-B
      ==============================================================================
      Testsuites.component-B.feature-B-1
      ==============================================================================
      Testsuites.component-B.feature-B-1.exercise-08-feature-B-1-test
      ==============================================================================
      Test Case exercise-08-feature-B-1-test :: exercise-08-feature-B-1-... | UNKNOWN |
      Variable '${teststring_common}' not found.
      ------------------------------------------------------------------------------
      Testsuites.component-B.feature-B-1.exercise-08-feature-B-1-test       | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-B.feature-B-1                                    | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-B.feature-B-2
      ==============================================================================
      Testsuites.component-B.feature-B-2.exercise-08-feature-B-2-test
      ==============================================================================
      Test Case exercise-08-feature-B-2-test :: exercise-08-feature-B-2-... | UNKNOWN |
      Variable '${teststring_common}' not found.
      ------------------------------------------------------------------------------
      Testsuites.component-B.feature-B-2.exercise-08-feature-B-2-test       | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-B.feature-B-2                                    | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites.component-B                                                | UNKNOWN |
      2 tests, 0 passed, 0 failed, 2 unknown
      ==============================================================================
      Testsuites.Exercise-08
      ==============================================================================
      Test Case exercise-08 :: exercise-08                                  | UNKNOWN |
      Variable '${teststring_common}' not found.
      ------------------------------------------------------------------------------
      Testsuites.Exercise-08                                                | UNKNOWN |
      1 test, 0 passed, 0 failed, 1 unknown
      ==============================================================================
      Testsuites                                                            | UNKNOWN |
      Suite setup failed:
      Unknown exception occurs. Details: Testsuite management - Loading configuration level 2 failed!
                       The variant 'default' is not defined in 'C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\robotframework-tutorial\900_Building_testsuites\exercise-08\config\exercise-08_variants.json'
      
      5 tests, 0 passed, 0 failed, 5 unknown
      ==============================================================================
      Debug:   C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\debug_log.log
      Output:  C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\output.xml
      Log:     C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\log.html
      Report:  C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\report.html

**Comment by test-fullautomation** [comment]
Please refer to #96

**Comment by namsonx** [comment]
Hello Thomas,

The logging function was updated in the pull-request #116 

Thank you,
Son

**Comment by test-fullautomation** [comment]
solved and already merged wiht #116
